vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -104,14 +105,17 @@ private String getAnnotationAttributes(VariantWithSamplesAndAnnotation variant)
         if (consequenceTypes != null) {
             List<String> consequences = new ArrayList<>();
             for (ConsequenceType consequenceType : consequenceTypes) {
-                Set<Integer> soTermList = consequenceType.getSoAccessions();
-                List<String> soTermStrList = new ArrayList<>();
-                if (soTermList != null) {
-                    soTermList.forEach(s -> soTermStrList.add(String.valueOf(s)));
+                List<String> soNameList = new ArrayList<>();
+                Set<Integer> soAccessions =  consequenceType.getSoAccessions();
+                if (soAccessions != null) {
+                    for (Integer integer : soAccessions) {
```

Comment:
This block can be replaced by a Stream operation using some lambdas:

List<String> soNameList = consequenceType.getSoAccessions().stream().flatMap(..).map(so -> ConsequenceTypeMappings.getSoName(so)).collect(.. ...

Reply:
Another option is to move that code to a separate function, to improve the readability. I don't have a strong preference by any of those.

Reply:
Actually, looking that later the List is 'collapsed' into a String, we can do it in the Lambda with a Collector, saving one step. Anyway, the separate function is not a bad idea either.